### ISSUE TITLE
The tokenizer for the double quotes symbol " is not assigning the corresponding ID from the vocabulary.

### ISSUE THREAD

**Issue by Yue327-27** [keras-team/keras-nlp#877]
When the tokenizer is applied to the `<|endoftext|>` not getting the id of the <|endoftext|>
```
tokenizer("<|endoftext|>")

<tf.Tensor: shape=(7,), dtype=int32, numpy=array([  27,   91,  437, 1659, 5239,   91,   29], dtype=int32)>
```

If this is normal?

**Comment by abheesht17** [comment]
This is, in fact, expected. We do not expect our tokenisers to map "special tokens" to their correct IDs. We do, however, add the correct tokens in our XXPreprocessor layers.

**Comment by Yue327-27** [comment]
Thanks @abheesht17 I will close this issue now

something useful for me 
https://github.com/keras-team/keras-nlp/blob/a5ca4eefb40baca647f73fb194d6280144e4eda0/keras_nlp/models/distil_bert/distil_bert_preprocessor.py#L139-L147

**Comment by Yue327-27** [comment]
Ah, also, may I ask a question, I found when I used `<|endoftext|>` in chatgpt. When it copied my question, it will stop generating when hitting `<|endoftext|>`, do you have any idea what is the logic? I understand it may stop because it seen the stop token, maybe the same for keras_nlp as well? any idea that we can avoid that?

**Comment by xihajun** [comment]
Also have the same question (trying to finetune a chatbot using gpt2)

```
preprocessor = keras_nlp.models.GPT2Preprocessor.from_preset("gpt2_base_en")
preprocessor("hello <|endoftext|>")
```

> output: {'token_ids': <tf.Tensor: shape=(1024,), dtype=int32, numpy=array([31373,  1279,    91, ..., 50256, 50256, 50256], dtype=int32)>,
 'padding_mask': <tf.Tensor: shape=(1024,), dtype=bool, numpy=array([ True,  True,  True, ..., False, False, False])>}

Does this suppose to be `([31373,  50256, ..., 50256, 50256, 50256], dtype=int32)>`?

**Comment by abheesht17** [comment]
@Yue327-27, yeah. 

Let me explain the flow a bit.
- In our sampler functions, we stop generating tokens when we hit the end token: https://github.com/keras-team/keras-nlp/pull/769.
- However, when no end token is specified, we generate text till max. length. Check this: https://github.com/keras-team/keras-nlp/blob/master/keras_nlp/samplers/sampler.py#L36. 
- The generate method for necessarily GPT-2 passes the end token here: https://github.com/keras-team/keras-nlp/blob/master/keras_nlp/models/gpt2/gpt2_causal_lm.py#L335. If, however, we do not pass end token here, we can have chatbot-like generation (generation till max. length). I believe work is underway: https://github.com/keras-team/keras-nlp/issues/844.

**Comment by abheesht17** [comment]
@xihajun, sorry if I misunderstood your question, but did you mean to ask why we do padding on the right instead of the left? I believe @chenmoneygithub tried some experiments with left padding and got bad quality results.

**Comment by abheesht17** [comment]
Oh, and @Yue327-27, if you want a temporary solution for this (since the PR for chatbot-like generation might take a while), in your `token_probability_fn`, you can "suppress" the end token ID. So, just set the logits for end token to a very small negative number (softmax normalisation will set the probability to 0).

**Comment by Yue327-27** [comment]
@abheesht17 Thank you for explaining the flow of the sampler functions and how the generate method for GPT-2 works. I appreciate the additional information about the ongoing work to modify the generate() method to support chatbot-like generation.

Also, thank you for the temporary solution for suppressing the end token ID in the token_probability_fn. I will keep my eyes on the chatbot-like generation as well :)
